recipes/libvpx/all/conanfile.py

```diff
@@ -148,53 +183,118 @@ def _configure_autotools(self):
-            # special case, as gcc/g++ is hard-coded in makefile, it implicitly assumes -lstdc++
-            autotools.link_flags.append("-stdlib=libc++")
-        autotools.configure(args=args, configure_dir=self._source_subfolder, host=False, build=False, target=False)
-        return autotools
+                    configure_args.append(f"--disable-{name}")
+
+        configure_command = " ".join(configure_args)
+        self.output.info(configure_command)
+        self.run(configure_command)
+
 
     def build(self):
-        self._patch_sources()
-        with tools.vcvars(self) if self._is_msvc else tools.no_op():
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
+        # Disable LTO for Visual Studio when CFLAGS doesn't contain -GL
+        if is_msvc(self):
+            cflags = " ".join(self.conf.get("tools.build:cflags", default=[], check_type=list))
+            lto = any(re.finditer("(^| )[/-]GL($| )", cflags))
+            if not lto:
+                self.output.info("Disabling LTO")
+                replace_in_file(self,
+                    os.path.join(self.source_folder, "build", "make", "gen_msvs_vcxproj.sh"),
+                    "tag_content WholeProgramOptimization true",
+                    "tag_content WholeProgramOptimization false",
+                )
+            else:
+                self.output.info("Enabling LTO")
 
-    def package(self):
-        self.copy(pattern="LICENSE", src=self._source_subfolder, dst="licenses")
-        with tools.vcvars(self) if self._is_msvc else tools.no_op():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-
-        if self._is_msvc:
-            # don't trust install target
-            tools.rmdir(os.path.join(self.package_folder, "lib"))
-            libdir = os.path.join(
-                "Win32" if self.settings.arch == "x86" else "x64",
-                "Debug" if self.settings.build_type == "Debug" else "Release",
+        # The compile script wants to use CC for some of the platforms (Linux, etc),
+        # but incorrectly assumes gcc is the compiler for those platforms.
+        # This can fail some of the configure tests, and -lpthread isn't added to the link command.
+        replace_in_file(self,
+            os.path.join(self.source_folder, "build", "make", "configure.sh"),
+            "  LD=${LD:-${CROSS}${link_with_cc:-ld}}",
+            """
+  LD=${LD:-${CROSS}${link_with_cc:-ld}}
+  if [ "${link_with_cc}" = "gcc" ]
+  then
+   echo "using compiler as linker"
+   LD=${CC}
+  fi
+"""
             )
-            self.copy("vpx*.lib", src=libdir, dst="lib")
+
+        autotools = Autotools(self)
+        # NOT USED # autotools.configure() # FIXME can use this if we can remove --host and --build flags
```

Comment:
A similar issue has been raised here by @sorny92 : https://github.com/conan-io/conan/issues/12431

Reply:
Yes there is a few issues open on that theme

Reply:
My plan is to finish this PR, and someone can cycle back and fix it up with the new stuff when it has landed.
I know Autotools probably has some work that needs to be done in Jan anyway.

Reply:
Nice job by the way! I had a bit of struggle trying to make a conan v2 version of Ffmpeg and your work may be very helpful!